### ISSUE TITLE
`eqx.filter_grad` Question

### ISSUE THREAD

**Issue by ASEM000** [patrick-kidger/equinox#189]
Hey Patrick, 
I wonder if this behavior is intended

```python
import equinox as eqx
import jax
import jax.numpy as jnp


@jax.grad 
def jax_grad(x):
    return x**2

@eqx.filter_grad
def eqx_grad(x):
    return x**2

jax_grad(1.), eqx_grad(1.)
# (DeviceArray(2., dtype=float32, weak_type=True), None)

```

```python
@jax.grad 
def jax_grad(x):
    return sum(x)**2

@eqx.filter_grad
def eqx_grad(x):
    return sum(x)**2

jax_grad([1.,2.,3.]), eqx_grad([1.,2.,3.])
#([DeviceArray(12., dtype=float32, weak_type=True),
#  DeviceArray(12., dtype=float32, weak_type=True),
#  DeviceArray(12., dtype=float32, weak_type=True)],
# [None, None, None])

```

**Comment by patrick-kidger** [comment]
Hey there. Yep, this is intended. By default, `eqx.filter_grad` will compute gradients with respect to floating-point JAX arrays. (And in particular not int arrays, Python floats or anything else.)

You can change this via `eqx.filter_grad(default=...)` if you want to. (Or on just a per-argument basis via `eqx.filter_grad(args=..., kwargs=...)`.)
